test/mock/core/network/beefy_protocol_mock.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "mock/core/network/protocol_base_mock.hpp"
+#include "network/protocols/beefy_protocol.hpp"
+
+#include <gmock/gmock.h>
+
+namespace kagome::network {
+
+  class BeefyProtocolMock : public ProtocolBaseMock, public BeefyProtocol {
```

Comment:
Does this work without `virtual`-inheritance?

Reply:
Yes, it does. ProtocolBase as common base class already is virtual.